rust/protocol/src/curve.rs

```diff
@@ -260,27 +336,48 @@ impl PrivateKey {
         self.calculate_signature_for_multipart_message(&[message], csprng)
     }
 
+    /// Calculate a signature for the multipart `message` given this private key.
     pub fn calculate_signature_for_multipart_message<R: CryptoRng + Rng>(
         &self,
         message: &[&[u8]],
         csprng: &mut R,
     ) -> Result<Box<[u8]>> {
         match self.key {
-            PrivateKeyData::DjbPrivateKey(k) => {
+            PrivateKeyData::Curve25519(k) => {
                 let private_key = curve25519::PrivateKey::from(k);
                 Ok(Box::new(private_key.calculate_signature(csprng, message)))
             }
         }
     }
 
-    pub fn calculate_agreement(&self, their_key: &PublicKey) -> Result<Box<[u8]>> {
+    /// Calculate a shared secret between this private key and the public key `their_key`.
+    pub fn calculate_agreement(
+        &self,
+        their_key: &PublicKey,
+    ) -> Result<[u8; curve25519::AGREEMENT_LENGTH]> {
```

Comment:
This is the place we talked about before: key agreement can't hardcode an output length, because the keys might (someday) not be curve25519 keys.